app/experimenter/nimbus-ui/src/hooks/useCommonForm.test.tsx

```diff
@@ -0,0 +1,119 @@
+      const { nameField, submitErrors } = overviewSetup();
+      const nameError = screen.getByText(submitErrors["name"][0]);
+      expect(nameField).toHaveClass("is-invalid");
+      expect(nameError).toBeInTheDocument();
+
+      await act(async () => {
+        fireEvent.change(nameField, {
+          target: { value: "abc" },
+        });
+        fireEvent.blur(nameField);
+      });
+
+      expect(nameField).toHaveClass("is-valid");
+      expect(nameError).not.toBeInTheDocument();
+    });
+
+    it("does not clear submit error onblur of unchanged field with error", async () => {
+      const { nameField, submitErrors } = overviewSetup();
+      const getHypothesisError = () =>
+        screen.queryByText(submitErrors["hypothesis"][0]);
+      const hypothesisField = screen.getByLabelText("Hypothesis");
+      expect(hypothesisField).toHaveClass("is-invalid");
+      expect(getHypothesisError()).toBeInTheDocument();
+
+      await act(async () => {
+        fireEvent.change(nameField, {
+          target: { value: "abc" },
+        });
+        fireEvent.blur(nameField);
+      });
+      expect(hypothesisField).toHaveClass("is-invalid");
+      expect(getHypothesisError()).toBeInTheDocument();
+    });
+
+    it("clears submit error onchange of multi select", () => {
+      // TODO
+    });
+  });
+
+  describe("is used on expected fields", () => {
+    describe("FormOverview", () => {
+      it("with experiment data", () => {
+        const { experiment } = mockExperimentQuery("boo");
+        render(<OverviewSubject {...{ experiment }} />);
+
+        overviewFieldNames.forEach((name) => {
+          if (name !== "application") {
+            expect(
+              screen.queryByTestId(`${name}-form-errors`),
+            ).toBeInTheDocument();
+            expect(screen.queryByTestId(name)).toBeInTheDocument();
+          }
+        });
+      });
+
+      it("without experiment data", async () => {
+        await act(async () => {
+          render(<OverviewSubject />);
+        });
+
+        overviewFieldNames.forEach((name) => {
+          if (name !== "publicDescription") {
+            expect(
+              screen.queryByTestId(`${name}-form-errors`),
+            ).toBeInTheDocument();
+            expect(screen.queryByTestId(name)).toBeInTheDocument();
+          }
+        });
+      });
+    });
+
+    it("FormMetrics", () => {
+      render(<MetricsSubject />);
+
+      metricsFieldNames.forEach((name) => {
+        expect(screen.queryByTestId(`${name}-form-errors`)).toBeInTheDocument();
+        // TODO
+        // expect(screen.queryByTestId(name)).toBeInTheDocument();
+      });
+    });
+
+    it("FormAudience", () => {
+      render(<AudienceSubject />);
+
+      audienceFieldNames.forEach((name) => {
+        expect(screen.queryByTestId(`${name}-form-errors`)).toBeInTheDocument();
+        expect(screen.queryByTestId(name)).toBeInTheDocument();
```

Comment:
What I really want to test for is:
1) if each field has a corresponding `<FormErrors {...{ name}} />`
2) if each field control calls `formControlAttrs` or `formSelectAttrs` with their respective `name`s

I tried mocking what's returned by `useCommonForm` inside each form and had trouble testing it that way with mock functions / a mock JSX element but I'll revisit this - otherwise we need that test span for L114, and on L115, technically you could manually add `name` to the input and `screen.queryByTestId(name)).toBeInTheDocument()` would pass without `formControlAttrs`.

Reply:
I've attempted this twice now and haven't really gotten anywhere so I made a follow up for it: https://jira.mozilla.com/browse/EXP-780